internal/token/migration_test.go

```diff
@@ -0,0 +1,161 @@
+package token_test
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd-oidc-brokers/internal/token"
+	"golang.org/x/crypto/scrypt"
+)
+
+func TestUseOldEncryptedToken(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		passwordFileExists          bool
+		newTokenFileExists          bool
+		oldEncryptedTokenFileExists bool
+
+		wantRet   bool
+		wantError bool
+	}{
+		"Success when both the password file and the new token file exist": {passwordFileExists: true, newTokenFileExists: true, wantRet: false},
+		"Success when old encrypted token file exists":                     {oldEncryptedTokenFileExists: true, wantRet: true, wantError: false},
+
+		"Error if only the password file exists":                                     {passwordFileExists: true, wantRet: false, wantError: true},
+		"Error if only the new token file exists":                                    {newTokenFileExists: true, wantRet: false, wantError: true},
+		"Error if neither the password file nor the old encrypted token file exists": {wantError: true},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			passwordPath := t.TempDir() + "/password"
+			tokenPath := t.TempDir() + "/token"
+			oldEncryptedTokenPath := t.TempDir() + "/oldtoken"
+
+			if tc.passwordFileExists {
+				err := os.WriteFile(passwordPath, []byte("password"), 0600)
+				require.NoError(t, err, "WriteFile should not return an error")
+			}
+			if tc.newTokenFileExists {
+				err := os.WriteFile(tokenPath, []byte("token"), 0600)
+				require.NoError(t, err, "WriteFile should not return an error")
+			}
+			if tc.oldEncryptedTokenFileExists {
+				err := os.WriteFile(oldEncryptedTokenPath, []byte("encryptedtoken"), 0600)
+				require.NoError(t, err, "WriteFile should not return an error")
+			}
+
+			got, err := token.UseOldEncryptedToken(passwordPath, tokenPath, oldEncryptedTokenPath)
+			if tc.wantError {
+				require.Error(t, err, "UseOldEncryptedToken should return an error")
+			} else {
+				require.NoError(t, err, "UseOldEncryptedToken should not return an error")
+				require.Equal(t, tc.wantRet, got, "UseOldEncryptedToken should return the expected value")
+			}
```

Comment:
I'm pretty sure we have addressed this before, so I'm not sure if got caught on a rebase or something, but we prefer returning early instead of an else here in these cases (it happens on other tests as well, so take a closer look at the other cases).

Reply:
done